connectors/sources/jira.py

```diff
@@ -0,0 +1,467 @@
+        self.is_cloud = self.configuration["is_cloud"]
+        self.host_url = self.configuration["host_url"]
+        self.ssl_disabled = self.configuration["ssl_disabled"]
+        self.certificate = self.configuration["ssl_ca"]
+        self.enable_content_extraction = self.configuration["enable_content_extraction"]
+        self.retry_count = self.configuration["retry_count"]
+        self.concurrent_downloads = self.configuration["concurrent_downloads"]
+
+        self.ssl_ctx = False
+        self.session = None
+        self.queue = MemQueue(
+            maxsize=QUEUE_SIZE, maxmemsize=QUEUE_MEM_SIZE * 1024 * 1024
+        )
+        self.fetchers = ConcurrentTasks(max_concurrency=MAX_CONCURRENCY)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for Jira
+
+        Returns:
+            dictionary: Default configuration.
+        """
+        return {
+            "is_cloud": {
+                "value": IS_CLOUD,
+                "label": "True if Jira Cloud, False if Jira Server",
+                "type": "bool",
+            },
+            "username": {
+                "value": "admin",
+                "label": "Jira Server username",
+                "type": "str",
+            },
+            "password": {
+                "value": "changeme",
+                "label": "Jira Server password",
+                "type": "str",
+            },
+            "service_account_id": {
+                "value": "me@example.com",
+                "label": "Jira Cloud service account id",
+                "type": "str",
+            },
+            "api_token": {
+                "value": "abc#123",
+                "label": "Jira Cloud API token",
+                "type": "str",
+            },
+            "host_url": {
+                "value": "http://127.0.0.1:8080",
+                "label": "Jira host url",
+                "type": "str",
+            },
+            "ssl_disabled": {
+                "value": DEFAULT_SSL_DISABLED,
+                "label": "Disable SSL verification. True if SSL is disabled else False",
+                "type": "bool",
+            },
+            "ssl_ca": {
+                "value": "",
+                "label": "SSL certificate",
+                "type": "str",
+            },
+            "enable_content_extraction": {
+                "value": DEFAULT_CONTENT_EXTRACTION,
+                "label": "Enable content extraction (true/false)",
+                "type": "bool",
+            },
+            "retry_count": {
+                "value": RETRIES,
+                "label": "Maximum retries for failed requests",
+                "type": "int",
+            },
+            "concurrent_downloads": {
+                "value": 50,
+                "label": "Number of concurrent downloads for fetching attachment content",
+                "type": "int",
+            },
+        }
+
+    def tweak_bulk_options(self, options):
```

Comment:
Isn't this function a validation function + assignment? I believe it could be partially moved into `_validate_configuration` (validation part), other part can go to initialiser.